### PR TITLE
DAOS-13974 control: Print actions in same order as JSON

### PR DESCRIPTION
**Pull request by kjacque**
The check query actions in the human-readable text output were in a different order than the JSON. The action index for the repair input is tied to the human-readable output order, so JSON users would have picked the wrong index.

Features: recovery

Required-githooks: true

### Before requesting gatekeeper:

* [x] Two review approvals and any prior change requests have been resolved.
* [x] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Comment by kjacque** [comment]
Test failures are all known issues:
- https://daosio.atlassian.net/browse/DAOS-13998 (since this test run, it was fixed in the parent branches)
- https://daosio.atlassian.net/browse/DAOS-13708 (rebuild timeout)

**Review by knard-intel — APPROVED** [review]

**Review by tanabarr — COMMENTED** [review]
shouldn't there be a test case to verify the change?

**Comment by kjacque** [comment]
> shouldn't there be a test case to verify the change?

Fair point, I'll add a test.

**Review by Nasf-Fan — APPROVED** [review]
I manually verified the patch, it works as expected.

File: src/control/lib/control/check.go [hunk 1]
```diff
@@ -414,9 +413,6 @@ func (r *SystemCheckReport) RepairChoices() []*SystemCheckRepairChoice {
 		}
 	}
 
-	sort.Slice(choices, func(i, j int) bool {
-		return choices[i].Action < choices[j].Action
-	})
```
Comment (mjmac) [inline src/control/lib/control/check.go:419]: Won't this result in the human-readable choices being kind of jumbled too?

Comment (kjacque) [reply src/control/lib/control/check.go:419]: The recommended action will be the first one. Does the order matter otherwise?

**Review by tanabarr — APPROVED** [review]

**Comment by kjacque** [comment]
Test failures are known issues:
- https://daosio.atlassian.net/browse/DAOS-13998
- https://daosio.atlassian.net/browse/DAOS-14085
- https://daosio.atlassian.net/browse/DAOS-11282
- https://daosio.atlassian.net/browse/DAOS-14027 (IOR and rebuild)

**Pull request merged**